wrapanapi/systems/openstack.py

```diff
@@ -583,15 +588,23 @@ class OpenstackSystem(System, VmMixin, TemplateMixin):
     can_suspend = True
     can_pause = True
 
-    def __init__(self, **kwargs):
-        self.keystone_version = kwargs.get('keystone_version', 2)
+    def __init__(self, tenant, username, password, auth_url, **kwargs):
+        self.keystone_version = kwargs.get("keystone_version")
+        if not self.keystone_version:
+            parsed_keystone_version = search(r'v([2-3])$', auth_url)
+            if parsed_keystone_version:
+                self.keystone_version = kwargs.get('keystone_version',
```

Comment:
you don't need kwards.get here since you already checked that it's not defined

Reply:
removed